Data/Graph/Inductive/PatriciaTree.hs

```diff
@@ -220,6 +224,25 @@ toAdj = concatMap expand . IM.toList
 fromAdj :: Adj b -> IntMap [b]
 fromAdj = IM.fromListWith addLists . map (second (:[]) . swap)
 
+data FromListCounting a = FromListCounting !Int !(IntMap a)
```

Comment:
I assume this is solely for the `!` usage?

Can you please add at least `Eq`, `Show` and `Read` instances for this?